cineast-api/src/main/java/org/vitrivr/cineast/api/APIEndpoint.java

```diff
@@ -97,103 +198,210 @@ private static Service dispatchService(boolean secure) {
             service.redirect.any("/mediaobject", "/");
             service.redirect.any("/mediaobject/:objectId", "/");
 
-        }
+        }*/
 
         /* Setup HTTP/RESTful connection (if configured). */
-        if (config.getEnableRest()) {
-            registerRoutes(service);
+        if (config.getEnableRest() || config.getEnableRestSecure()) {
+            this.registerRoutes(config, service);
         }
 
         /* Register a general exception handler. TODO: Add fine grained exception handling. */
-        service.exception(Exception.class, (exception, request, response) -> {
-            exception.printStackTrace();
-            LOGGER.error(exception);
+        service.exception(Exception.class, (ex, ctx) -> {
+            ex.printStackTrace();
+            LOGGER.error(ex);
         });
 
         /* Configure the result after processing was completed. */
-        service.after((request, response) -> {
-            response.header("Access-Control-Allow-Origin", "*");
-            response.header("Access-Control-Allow-Headers", "*");
+        service.after(ctx -> {
+            ctx.header("Access-Control-Allow-Origin", "*");
+            ctx.header("Access-Control-Allow-Headers", "*");
         });
 
+        /* Start javalin */
+        try {
+            if (port > 0) {
+                service.start(port);
+            } else {
+                service.start();
+            }
+        } catch(Exception ex) {
+            LOGGER.log(Level.FATAL, "Failed to start HTTP endpoint due to an exception. Cineast will shut down now!", ex);
+            System.exit(100);
+        }
+
         return service;
     }
 
+    private int validatePort(boolean secure, APIConfig config) {
+        int port = secure ? config.getHttpsPort() : config.getHttpPort();
+        if(port <= 0 || port >= 65535) {
+            LOGGER.warn("The specified port {} is not valid. Fallback to default port.", port);
+            return -1;
```

Comment:
In my opinion, it would be more readable to have a constant for 'use default port'. Subsequently, in line 222, check explicitly for that default port constant.